changelog.d/13174.bugfix

```diff
@@ -0,0 +1 @@
+Fix `_get_state_map_for_room` in third_party_rules to stop breaking when we have in our db an ill-formed event *and* at least one `on_new_event` callback.
```

Comment:
I think this is maybe a little too specific- Perhaps something more along the lines of  "Use the more robust get_current_state to implement _get_state_map_for_room to avoid breakages." would be better?